metarecord/tests/test_api.py

```diff
@@ -1852,6 +1853,49 @@ def test_function_post_when_not_allowed(post_function_data, user_api_client):
     assert expected_error in response.data['non_field_errors']
 
 
+@pytest.mark.django_db
+def test_function_post_new_when_existing_function(rf, function, phase, action, record, user_api_client):
+    set_permissions(user_api_client, Function.CAN_EDIT)
+    classification = function.classification
+    classification.pk = None
+    classification.save()
+
+    function.refresh_from_db()
+
+    new_classification = Classification.objects.latest_version().get(code=function.classification.code)
+    post_data = {'classification': {
+        'id': new_classification.uuid.hex,
+        'version': new_classification.version,
+    }}
+
+    response = user_api_client.post(FUNCTION_LIST_URL, data=post_data)
+
+    class view():
+        action = "create"
+
+    dummy_request = rf.get(get_function_detail_url(function))
+    dummy_request.user = user_api_client.user
+    dummy_context = {"view": view(), "request": dummy_request}
+    function_data = FunctionTestDetailSerializer(function, context=dummy_context).data
+    response_data_json = response.json()
+    for attr in ['modified_at', 'created_at', 'actions']:
+        response_data_json['phases'][0].pop(attr)
+        function_data['phases'][0].pop(attr)
+    assert response_data_json['phases'][0] == function_data['phases'][0]
+    response_data_json.pop('phases')
+    function_data.pop('phases')
+
+    new_phase = Function.objects.filter(classification=new_classification).latest_version().first().phases.first()
+    assert (new_phase.uuid == phase.uuid)
+    assert new_phase.actions.first().uuid == phase.actions.first().uuid
+    assert new_phase.actions.first().records.first().uuid == phase.actions.first().records.first().uuid
```

Comment:
Phases, actions and records should have unique uuid. The uuid non-uniqueness affects only classifications and functions which are versioned.

Reply:
Ah got it mixed up. Will do.